### PR TITLE
Use NullLoggerFactory.Instance in DefaultServiceResolver

### PR DESCRIPTION
**Pull request by adammorr**

### CODE REVIEW

**Comment by adammorr** [comment]
@slang25 @maurofranchi @martincostello could I have a reapproval please...I added some tests to make codecov happy

File: tests/JustSaying.UnitTests/Fluent/WhenUsingDefaultServiceResolver.cs [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        [Fact]
+        public void ShouldResolveIMessagingConfig()
+        {
+            _sut.ResolveService<IMessagingConfig>().ShouldBeOfType<MessagingConfig>();
+        }
+
+        [Fact]
+        public void ShouldResolveIMessageSerializationFactory()
+        {
+            _sut.ResolveService<IMessageSerializationFactory>().ShouldBeOfType<NewtonsoftSerializationFactory>();
```
Comment (slang25) [inline tests/JustSaying.UnitTests/Fluent/WhenUsingDefaultServiceResolver.cs]: Pedantic: either the assert should be "not null", or the test name should say what the default type gets resolved

Comment (adammorr) [reply tests/JustSaying.UnitTests/Fluent/WhenUsingDefaultServiceResolver.cs]: You are so damn picky Lang!

Comment (adammorr) [reply tests/JustSaying.UnitTests/Fluent/WhenUsingDefaultServiceResolver.cs]: Updated @slang25 👍

Comment (slang25) [reply tests/JustSaying.UnitTests/Fluent/WhenUsingDefaultServiceResolver.cs]: Guilty 😄

**Pull request merged**